packages/sol-compiler/src/utils/compiler.ts

```diff
@@ -437,3 +474,47 @@ export function getDependencyNameToPackagePath(
     });
     return dependencyNameToPath;
 }
+
+/**
+ * Extract the solidity version (e.g., '0.5.9') from a solc version (e.g., `0.5.9+commit.34d3134f`).
+ */
+export function getSolidityVersionFromSolcVersion(solcVersion: string): string {
+    const m = /(\d+\.\d+\.\d+)\+commit\.[a-f0-9]{8}/.exec(solcVersion);
```

Comment:
Not sure if relevant, but we may want to handle uppercase hex with `[a-fA-F0-9]`. We may also be able to replace the hex expression with `[[:xdigit:]]`.